docs/usage/rest.md

```diff
@@ -2,16 +2,78 @@
+## Methods
+
+The `RestClient` offers the 4 core request methods: `get`, `delete`, `post`, and `put`, . These methods each take in an associated `<method>RequestParams` object as their argument, each with a set of required and optional parameters. The tables below outline these params objects.
+
+**Get and Delete:**
+
+- The `get` method accepts a `GetRequestParams` object, which accepts the following parameters: `path`, `type`, `data`, `query`, `extraHeaders`, and `tries`.
+- The `delete` method accepts a `DeleteRequestParams` object, **which has an identical strucuture to `GetRequestParams`**.
+-
+- **`GetRequestParams` / `DeleteRequestParams`:**
+  | Parameter | Type | Required? | Default Value | Notes |
+  | -------------- | ----------------------------------- | :-------: | :-----------: | ---------------------------------------------------------------------------------------- |
+  | `path` | `string` | True | none | The requested API endpoint path |
+  | `data` | `Record<string, unknown> \| string` | False | none | The body of the request |
+  | `type` | `DataType` | False | none | The type of data being sent in the body of the request (`JSON`, `GraphQL`, `URLEncoded`) |
+  | `query` | `Record<string, string \| number>` | False | none | An optional query object to be appended to the request |
+  | `extraHeaders` | `Record<string, string \| number>` | False | none | Any additional headers you want to send with your request |
+  | `tries` | `number` | False | `1` | The maximum number of times to retry the request _(must be >= 0)_ |
+
+**Post and Put:**
+
+- The `post` method accepts a `PostRequestParams` object, which differs from `get` in that both the `type` and `data` parameters are required, in addition to the `path`.
+- The `put` method accepts a `PutRequestParams` object, **which has an identical structure to `PostRequestParams`**.
+-
+- **`PostRequestParams` / `PutRequestParams`:**
+  | Parameter | Type | Required? | Default Value | Notes |
+  | -------------- | ----------------------------------- | :-------: | :-----------: | ---------------------------------------------------------------------------------------- |
+  | `path` | `string` | True | none | The requested API endpoint path |
+  | `data` | `Record<string, unknown> \| string` | True | none | The body of the request |
+  | `type` | `DataType` | True | none | The type of data being sent in the body of the request (`JSON`, `GraphQL`, `URLEncoded`) |
+  | `query` | `Record<string, string \| number>` | False | none | An optional query object to be appended to the request |
+  | `extraHeaders` | `Record<string, string \| number>` | False | none | Any additional headers you want to send with your request |
+  | `tries` | `number` | False | `1` | The maximum number of times to retry the request _(must be >= 0)_ |
+
+## Usage Examples:
+
+We can run the code below in any endpoint where we have access to the request and response objects.
+
+**Perform a `get` request:**
```

Comment:
Nit: use GET / POST rather than get / post?

Reply:
Fair!

Reply:
I think I went for the lowercase because it's the literal method (`client.get`) but using the uppercase is probably more clear and more conventional.